docs/release-notes/2025-08-08-faststore-store-coupons.md

```diff
@@ -0,0 +1,36 @@
+---
+title: "FastStore: Coupons persist through store cart validation"
+slug: "2025-08-08-faststore-store-coupons"
+type: fixed
+excerpt: "Coupons stay applied after store cart updates"
+createdAt: "2025-08-08T09:50:00.214Z"
+updatedAt: "2025-08-08T10:00:00.214Z"
+hidden: false
+---
+
+Applied coupons now remain in the store cart after any cart interaction, such as refreshing the page or adding new items.
+
+## What has changed?
+
+Previously, applied coupons in the cart would disappear after a page refresh or any interaction.
+This happened because the `marketingData` object, which holds coupon information and other session data (like UTMs), was being overwritten during cart interaction, causing applied coupons to be deleted from the cart.
+
+Now, when a user refreshes their cart or adds a new item, existing coupon information within `marketingData` is combined with any new session data instead of being overwritten. This ensures that the previously applied coupons and other `marketingData` properties like UTMs are preserved.
+
+## Why did we make this change?
+
+Losing applied coupon codes during cart validation was a point of friction for customers, which could lead to abandoned carts and lost sales. This fix ensures `marketingData` is combined rather than overwritten. The update eliminates this friction point, contributing to a better customer journey.
```

Comment:
```suggestion
Losing applied coupon codes during cart validation could lead to abandoned carts and lost sales. This fix ensures `marketingData` is combined rather than overwritten. The update eliminates this friction point, contributing to a better customer journey.
```